File: python/single_threaded.py
import os
import sys


def readWriteFile(file_path, output_path):

    print(f"Reading file...")

    with open(file_path, "r") as file:
        lines = file.readlines()

    print(f"Writing file...")
    with open(output_path, "w") as file:
        for line in lines:
            file.write(line)

    print("Done!")


def main():
    print("\n\n-----------------------------------")
    print("\nPython Single Threaded Script")
    print("\n-----------------------------------\n\n")

    base_path = sys.argv[1] if len(sys.argv) > 1 else "../"
    input_file = sys.argv[2] if len(sys.argv) > 2 else "biggerSample.txt"
    file_path = os.path.join(base_path, input_file)
    output_file = os.path.join(base_path, "data", "outputPython.txt")

    readWriteFile(file_path, output_file)


if __name__ == "__main__":
    main()

File: rust/src/bin/single_threaded.rs
use std::env;
use std::fs::{self, File};
use std::io::{self, Read, Write};
use std::path::Path;

fn read_write_file(input_path: &Path, output_path: &Path) -> io::Result<()> {
    let mut input_file = File::open(input_path)?;
    let mut output_file = File::create(output_path)?;

    let mut buffer = String::new();
    println!("Reading file...");
    input_file.read_to_string(&mut buffer)?;

    println!("Writing file...");
    output_file.write_all(buffer.as_bytes())?;

    println!("Done!");

    Ok(())
}

fn main() -> io::Result<()> {
    println!("\n\n-----------------------------------");
    println!("\nRust Single Threaded Script");
    println!("\n-----------------------------------\n\n");

    let args: Vec<String> = env::args().collect();

    let base_path = if args.len() > 1 { &args[1] } else { "../../../" };

    let input_file = if args.len() > 2 {
        &args[2]
    } else {
        "data/biggerSample.txt"
    };

    let input_file_path = Path::new(base_path).join(input_file);
    let output_file_path = Path::new(base_path).join("data").join("outputRust.txt");

    read_write_file(&input_file_path, &output_file_path)?;

    Ok(())
}
